src/pages/RFOX/components/Stake/StakeConfirm.tsx

```diff
@@ -163,6 +180,29 @@ export const StakeConfirm: React.FC<StakeConfirmProps & StakeRouteProps> = ({
     [approvalTx?.status],
   )
 
+  const hasEnoughStakingAssetFeeBalance = useMemo(() => {
+    // Staking asset fee asset still loading, assume enough balance not to have a flash of error state on first render
+    if (!stakingAssetFeeAsset) return true
+    if (bnOrZero(stakingAmountCryptoPrecision).isZero()) return true
+    if (bnOrZero(stakingAssetFeeAssetBalanceCryptoPrecision).isZero()) return false
+
+    const fees = approvalFees || stakeFees
```

Comment:
question: should we be summing the approval fee with the staking fee so a user is never able to get into a situation where they only have enough balance to approve but after that not enough to stake?

Reply:
e.g 
```typescript
const totalNetworkFeeCryptoBaseUnit = bnOrZero(approvalFees?.totalNetworkFeeCryptoBaseUnit)
  .plus(bnOrZero(stakeFees?.totalNetworkFeeCryptoBaseUnit))
  ```

Reply:
Was originally going to say absolutely! Unfortunately, we can't get Tx fees if an approval is required, because getting Tx fees means simulating the Tx, and the Tx would revert on approval needed.
So `bnOrZero(stakeFees)` would always evaluate to 0 in `bnOrZero(approvalFees).plus(bnOrZero(stakeFees))` if an approval is required

Reply:
Makes sense, actually we should comment that!

Reply:
```suggestion
    // Unfortunately, we can't get Tx fees if an approval is required, because getting Tx fees means simulating the Tx, and the Tx would revert on approval needed.
    // So bnOrZero(stakeFees?.totalNetworkFeeCryptoBaseUnit) would always evaluate to 0 in the expression above, if an approval is required.
    const fees = approvalFees || stakeFees
```